parser/src/elements/indents/bullet_list.rs

```diff
@@ -0,0 +1,274 @@
+    CannotConvertSymbol,
+}
+
+pub(crate) enum EntryToken {
+    Id(String),
+    Keyword(BulletListEntryKeyword),
+    Heading(Vec<Inline>),
+    Body(Vec<Block>),
+    Attributes(String),
+}
+
+const STAR_ENTRY_START: &[SymbolKind] = &[
+    SymbolKind::Newline,
+    SymbolKind::Star,
+    SymbolKind::Whitespace,
+];
+const MINUS_ENTRY_START: &[SymbolKind] = &[
+    SymbolKind::Newline,
+    SymbolKind::Minus,
+    SymbolKind::Whitespace,
+];
+const PLUS_ENTRY_START: &[SymbolKind] = &[
+    SymbolKind::Newline,
+    SymbolKind::Plus,
+    SymbolKind::Whitespace,
+];
+
+const STAR_SUB_ENTRY_START: &[SymbolKind] = &[
+    SymbolKind::Newline,
+    SymbolKind::Whitespace,
+    SymbolKind::Whitespace,
+    SymbolKind::Star,
+    SymbolKind::Whitespace,
+];
+const MINUS_SUB_ENTRY_START: &[SymbolKind] = &[
+    SymbolKind::Newline,
+    SymbolKind::Whitespace,
+    SymbolKind::Whitespace,
+    SymbolKind::Minus,
+    SymbolKind::Whitespace,
+];
+const PLUS_SUB_ENTRY_START: &[SymbolKind] = &[
+    SymbolKind::Newline,
+    SymbolKind::Whitespace,
+    SymbolKind::Whitespace,
+    SymbolKind::Plus,
+    SymbolKind::Whitespace,
+];
+
+impl ElementParser for BulletList {
+    type Token<'a> = self::BulletListEntry;
+
+    fn tokenize<'i>(
+        input: &mut unimarkup_commons::scanner::SymbolIterator<'i>,
+    ) -> Option<crate::TokenizeOutput<Self::Token<'i>>> {
+        let mut tokens = Vec::new();
+
+        // `[1..]` to strip newline match for list start
+        while input.matches(&STAR_ENTRY_START[1..])
+            || input.matches(&MINUS_ENTRY_START[1..])
+            || input.matches(&PLUS_ENTRY_START[1..])
+        {
+            match BulletListEntry::tokenize(input) {
+                Some(entry_tokens) => {
+                    let Block::BulletListEntry(entry) =
+                        BulletListEntry::parse(entry_tokens.tokens)?.pop()?
+                    else {
+                        return None;
+                    };
+
+                    tokens.push(entry);
+                }
+                None => break,
+            }
+        }
+
+        if tokens.is_empty() {
+            return None;
+        }
+
+        Some(crate::parser::TokenizeOutput { tokens })
+    }
+
+    fn parse(input: Vec<Self::Token<'_>>) -> Option<crate::elements::Blocks> {
+        let mut list = Self {
+            id: String::new(),
+            entries: Vec::with_capacity(input.len()),
+        };
+
+        for entry in input {
+            list.entries.push(entry);
+        }
+
+        Some(vec![Block::BulletList(list)])
+    }
+}
+
+impl ElementParser for BulletListEntry {
+    type Token<'a> = self::EntryToken;
+
+    fn tokenize<'i>(
+        input: &mut unimarkup_commons::scanner::SymbolIterator<'i>,
+    ) -> Option<crate::TokenizeOutput<Self::Token<'i>>> {
+        let entry_keyword = BulletListEntryKeyword::try_from(input.next()?).ok()?;
+
+        if SymbolKind::Whitespace != input.next()?.kind {
```

Comment:
I would reverse this, find it easier to read: 

```suggestion
        if input.next()?.kind != SymbolKind::Whitespace {
```